articles/active-directory/manage-apps/disable-user-sign-in-portal.md

```diff
@@ -33,6 +33,10 @@ It's easy to disable an enterprise application so no users can sign in to it in
 1. On the ***appname*** - **Properties** pane, select **No** for **Enabled for users to sign-in?**.
 1. Select the **Save** command.
 
+## Disable unlisted Enterprise apps
+
+To disable apps that uses OAuth2 implicit flow, the tenant admin must manually create and disable the ServicePrincipal in their tenant using the [AzureAD PowerShell cmdlet](https://docs.microsoft.com/en-us/powershell/module/azuread/New-AzureADServicePrincipal?view=azureadps-2.0). For e.g. if the tenant admin wants to disable a public client app (SPA, mobile, etc.) that calls Microsoft Graph, then the Service Principal would need to be created and disabled using PowerShell.
```

Comment:
We should include a sample script demonstrating exactly how to do this:

````powershell
# The AppId of the app to be disabled
$appId = "e34cf59d-0121-4c10-bda8-149b2172b657"

# Check if a service principal already exists for the app
$servicePrincipal = Get-AzureADServicePrincipal -Filter "appId eq '$appId'"
if ($servicePrincipal) {
    # Service principal exists already, disable it
    Set-AzureADServicePrincipal -ObjectId $servicePrincipal.ObjectId -AccountEnabled $false
} else {
    # Service principal does not yet exist, create it and disable it at the same time
    $servicePrincipal = New-AzureADServicePrincipal -AppId $appId -AccountEnabled $false
}

Reply:
code snippet is now added.